controllers/user.py

```diff
@@ -1034,4 +1037,69 @@ def custom_friend():
                 allowed=allowed_custom_friends,
                 total_referrals=total_referrals)
 
+# ----------------------------------------------------------------------------
+def get_friend_list():
+    """
+        Gets the friend list of a specific user and compares with the friend 
+        list of the logged in user.
+    """
+
+    profile_user_id = request.vars["user_id"]
+    user_friends, _ = utilities.get_friends(session.user_id)
+    profile_friends, _ = utilities.get_friends(profile_user_id)
```

Comment:
According to the current logic, the view friend list button is not displayed on users' own profile because they can anyway go to `Your Friends` and see their friend list.